controlplane/kubeadm/controllers/status.go

```diff
@@ -69,6 +70,7 @@ func (r *KubeadmControlPlaneReconciler) updateStatus(ctx context.Context, kcp *c
 	// This only gets initialized once and does not change if the kubeadm config map goes away.
 	if status.HasKubeadmConfig {
 		kcp.Status.Initialized = true
+		conditions.MarkTrue(kcp, controlplanev1.AvailableCondition)
```

Comment:
I'm not sure we should tie `AvailableCondition` to whether there is a kubeadmconfig. Should this check the `/healthz/ready` endpoint for the apiserver instead?

Reply:
Both? :)

Reply:
Both definitely doesn't hurt :)